content/blogs/2023-08-16-datamesh.md

```diff
@@ -0,0 +1,161 @@
+
+To optimize performances and avoid resource overutilization, the data collection process is performed in batches of five, with parallelization using Kestra's workflow capabilities.
+
+To handle large volumes of collected data efficiently, the flow includes segmentation of API calls and transformation operations on historical data. This approach helps dilute the data volume and improve performance. 
+
+The received JSON data from Qualtrics is unzipped within a Docker environment. Python scripts are then used to transform the JSON, applying recursive functions for standardization, field transposition/renaming, and pivoting surveys into a generic key/value format for storage in BigQuery.
+
+---
+
+### Custom Plugins: From Collection to Centralization
+They aimed to further simplify the ingestion process and eliminate the burden of data loading. To meet this need, they developed a custom Kestra plugin, which automated several essential steps in their data handling process:
+
+- **Archiving incoming data**: The plugin archives incoming data to a Google Cloud Storage Bucket, providing a reliable backup.
+
+- **Validating data**: To prevent bad data ingestion, the plugin technically validates the data using an Avro schema. This ensures the appropriate typing (integer, date, etc.) of all incoming data.
+
+- **Data versioning**: The plugin creates a new version of data if a breaking change in the schema occurs, ensuring continuity and consistency.
+
+- **Appending technical columns**: Technical columns such as data load time, execution ID are appended by the plugin, allowing full lineage tracing in the data warehouse.
+
+- **Loading data into a temporary table**: Before final processing, the data is first loaded into a temporary table.
+
+- **Quality checks**: The plugin validates data quality based on specific business rules (upper/lower bound checks, key validation with referential, deduplication, etc.) before loading into the Operational Data Store (ODS).
+
+- **ODS loading**: The cleaned, validated data is finally loaded into the ODS, providing a mirror image of data from the incoming system.
+
+Data transfer was set up via HTTPS directly to the Kestra API, reducing dependencies. In most cases, a simple curl command was used for triggering ingestion, or a straightforward HTTP client was developed to reach the Kestra API.
+
+---
+
+### Enhance Supply with Report Automation ###
+To support charitable initiatives, each store allocates a portion of the margin to be donated to program-affiliated associations. The task to calculate these sums, inform store leaders, and provide payment details is managed by Kestra.
+
+The data platform team used Kestra's capabilities to develop a dedicated workflow to handle this process. This workflow calculates the precise amounts to be allocated based on store performance metrics and other relevant data. The resulting sums are then prepared for distribution to each Leader Perf at the respective stores.
```

Comment:
```suggestion
The data platform team used Kestra's capabilities to develop a dedicated workflow to handle this process. This workflow calculates the precise amounts to be allocated based on store performance metrics and other relevant data. The resulting sums are then prepared for distribution to the respective stores.
```
either explain what is Leader Perf or don't mention it

it confused me a lot :)